polkadot/node/core/approval-voting/src/persisted_entries.rs

```diff
@@ -214,7 +215,14 @@ impl ApprovalEntry {
 			},
 		};
 
-		self.tranches[idx].assignments.push((validator_index, tick_now));
+		if !is_duplicate ||
```

Comment:
I am not getting the !is_duplicate part. Can you give me your line of reasoning here, that would help I think. Thanks!